### PR TITLE
Improve documentation of `pub token` and subcommands

### PR DESCRIPTION
**Pull request by sigurdm**
Fixes: https://github.com/dart-lang/pub/issues/3111

### CODE REVIEW

File: lib/src/command/token.dart [hunk 1]
```diff
@@ -12,8 +13,17 @@ class TokenCommand extends PubCommand {
...
   @override
-  String get description =>
-      'Manage authentication tokens for hosted pub repositories.';
+  String get description => '''
+Manage authentication tokens for hosted pub repositories.
+
+The tokens will be used for authorizing both when retrieving dependencies and
+for publishing.
+
+Tokens are stored in ${tokenStore.tokensFile}.
```
Comment (jonasfj) [inline lib/src/command/token.dart:22]:
```suggestion
Tokens are stored in `${tokenStore.tokensFile}`.
```

File: lib/src/command/token_add.dart [hunk 2]
```diff
@@ -18,12 +19,23 @@ class TokenAddCommand extends PubCommand {
...
+
+The token will be used for authorizing against <hosted-url> both when
+retrieving dependencies and for publishing.
+
+Tokens are stored in ${tokenStore.tokensFile}.
+
+This command will prompt for the secret token over stdin. (Unless `--env-var` is
+used).
+
+For interactive auhtorization against pub.dev, use `$topLevelProgram pub login`.''';
```
Comment (jonasfj) [inline lib/src/command/token_add.dart:33]:
```suggestion
For interactive authorization against pub.dev, use `$topLevelProgram pub login`.''';
```

**Review by jonasfj — APPROVED** [review]

File: lib/src/command.dart [hunk 3]
```diff
@@ -37,7 +37,17 @@ const pubCommandAliases = {
   'upgrade': ['update'],
 };
 
-final lineLength = stdout.hasTerminal ? stdout.terminalColumns : 80;
+final lineLength = _lineLength();
+
+int _lineLength() {
+  final fromEnv = Platform.environment['_PUB_TEST_TERMINAL_COLUMNS'];
```
[the remark this answers is not in the document]
Comment (sigurdm) [reply lib/src/command.dart:43]: > Should we consider ignoring whitespace when we compare the golden files?
or just line breaks...

I considered it - but would really prefer that we get notified even of white-space changes to output behavior.

File: lib/src/command/token.dart [hunk 4]
```diff
@@ -12,8 +13,17 @@ class TokenCommand extends PubCommand {
...
+
+For interactive auhtorization against pub.dev, use `$topLevelProgram pub login`.''';
```
[the remark this answers is not in the document]
Comment (sigurdm) [reply lib/src/command/token.dart:24]: Thanks!

**Pull request merged**